libs/domain/user/address-list-item/address-list-item.component.ts

```diff
@@ -112,13 +84,52 @@ export class UserAddressListItemComponent
     </div>`;
   }
 
-  protected emitEvent(event: string, address: Address): void {
-    this.dispatchEvent(
-      new CustomEvent(event, {
-        bubbles: true,
-        composed: true,
-        detail: { address },
-      })
-    );
+  protected renderActions(): TemplateResult | void {
+    const address = this.$address();
+    const { editable, removable } = this.$options();
+    if (!address || (!editable && !removable)) return;
+
+    return html`<div class="controls">
+      ${this.renderEditTrigger()} ${this.renderRemoveTrigger()}
+    </div>`;
+  }
+
+  protected renderEditTrigger(): TemplateResult | void {
+    const address = this.$address();
+    const { editable, editTarget } = this.$options();
+    if (!address || !editable) return;
+
+    if (editTarget === EditTarget.Link) {
+      return html`
+        <oryx-icon-button outline>
+          <a href=${this.editLink()}>
+            <oryx-icon .type=${IconTypes.Edit}></oryx-icon>
+          </a>
+        </oryx-icon-button>
+      `;
+    }
+
+    return html`
+      <oryx-icon-button>
+        <button aria-label=${i18n('user.address.edit')} @click=${this.onEdit}>
+          <oryx-icon .type=${IconTypes.Edit}></oryx-icon>
+        </button>
+      </oryx-icon-button>
+    `;
+  }
+
+  protected renderRemoveTrigger(): TemplateResult | void {
+    if (!this.$options().removable) return;
+    return html`
+      <oryx-user-address-remove
+        .addressId=${this.$address()?.id}
+      ></oryx-user-address-remove>
+    `;
+  }
+
+  protected onEdit(): void {
+    const address = this.$address();
+    if (!address?.id) return;
```

Comment:
Should we throw some error (at least show the notification) in case if address does not have id?

Reply:
refactored